### PR TITLE
docs(zh-cn): update input/password with fully translated content

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

update input/password with fully translated content

### CODE REVIEW

File: files/zh-cn/web/html/element/input/password/index.md [hunk 1]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
+
+### pattern
+
+如果指定了 `pattern` 属性，为了使 {{htmlattrxref("value")}} 通过[约束验证](/zh-CN/docs/Web/HTML/Constraint_validation)，必须满足该属性给定的正则表达式。它必须是 {{jsxref("RegExp")}} 类型的有效 JavaScript 正则表达式，并且已在我们的[正则表达式指南](/zh-CN/docs/Web/JavaScript/Guide/Regular_Expressions)中进行了说明；在编译正则表达式时指定了 `'u'` 标志，因此该模式被视为 Unicode 代码点的序列，而不是 ASCII。模式文本周围无需指定正斜杠。
+
+如果没有指定模式，或指定了无效的模式，则不会应用任何正则表达式，且该属性将被完全忽略。
+
+> **备注：** 使用 {{htmlattrxref("title", "input")}} 属性来指定大多数浏览器将作为工具提示的文本，以解释匹配该模式的要求是什么。还应该在附近包括其他解释性的文本。
+
+强烈建议在密码输入中使用模式，以帮助确保用户选择和使用有效的密码，并使用广泛的字符类别。通过一个模式，你可以规定大小写规则，要求使用一些数字和/或标点符号，等等。参阅[验证](#验证)一节以获取详细说明和示例。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:50]:
```suggestion
强烈建议在密码输入中使用模式，以帮助确保用户选择和使用有效的密码，并使用广泛的字符类别。通过一个模式，你可以规定大小写规则、要求使用一些数字和/或标点符号，等等。参阅[验证](#验证)一节以获取详细说明和示例。
```

File: files/zh-cn/web/html/element/input/password/index.md [hunk 2]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
+
+### readonly
+
+一个布尔属性，如果存在，则表示该字段不能由用户编辑。但是，仍可以通过 JavaScript 代码直接设置 {{domxref("HTMLInputElement","HTMLInputElement.value")}} 的 `value` 属性来更改。
+
+> **备注：** 因为只读字段不能有值，所以 `required` 对指定了 `readonly` 属性的输入没有任何影响。
+
+### size
+
+`size` 属性是一个数字值，指示输入字段应有多少个字符宽。该值必须是一个大于零的数字，默认值是 20。由于字符宽度各不相同，这可能是也可能不是精确的，不应依赖于此；结果输入可能比指定的字符数更窄或更宽，这取决于字符和字体（{{cssxref("font")}} 使用中的设置）。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:68]:
```suggestion
`size` 属性是一个数字值，指示输入字段应有多少个字符宽。该值必须是一个大于零的数字，默认值是 20。由于字符宽度各不相同，这可能是也可能不是精确的，不应依赖于此；结果输入可能比指定的字符数更窄或更宽，这取决于字符和字体（使用的 {{cssxref("font")}} 设置）。
```

File: files/zh-cn/web/html/element/input/password/index.md [hunk 3]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
-The {{htmlattrxref("value", "input")}} attribute contains a {{domxref("DOMString")}} whose value is the current contents of the text editing control being used to enter the password. 如果用户还没有输入任何内容，则此值为空字符串（“”）。If the {{htmlattrxref("required")}} property is specified, then the password edit box must contain a value other than an empty string to be valid.
+{{htmlattrxref("value", "input")}} 属性包含了用于输入密码的文本编辑控件的当前内容的字符串。如果用户还没有输入任何内容，则此值为空字符串（`""`）。如果指定了 {{htmlattrxref("required")}} 属性，则密码编辑框必须包含一个非空的有效值。
+
+如果指定了 {{htmlattrxref("pattern", "input")}} 属性，则 `password` 中的内容只有通过验证才有效；参见[验证](#验证)一节以获得更多信息。
+
+> **备注：** 换行（U+000A）和回车（U+000D）符不允许在 `password` 中出现。当对密码控件设置值时，换行和回车符都会从值中删除。
+
+## 额外属性
+
+除了所有 {{HTMLElement("input")}} 元素支持的属性外，密码域输入还支持以下属性：
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:28]:
```suggestion
除了所有 {{HTMLElement("input")}} 元素支持的属性外，密码字段输入还支持以下属性：
```

File: files/zh-cn/web/html/element/input/password/index.md [hunk 4]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
+  maxlength="8"
+  size="8" />
 ```
 
 {{EmbedLiveSample("设置长度要求", 600, 40)}}
 
 ### 选中字符
 
-As with other textual entry controls, you can use the {{domxref("HTMLInputElement.select", "select()")}} method to select all the text in the password field.
+与其他文本输入控件一样，你可以使用 {{domxref("HTMLInputElement.select", "select()")}} 方法来选择密码域的所有字符。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:149]:
```suggestion
与其他文本输入控件一样，你可以使用 {{domxref("HTMLInputElement.select", "select()")}} 方法来选择密码字段的所有字符。
```

File: files/zh-cn/web/html/element/input/password/index.md [hunk 5]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
 
-- {{htmlattrdef("disabled")}}
-  - : 此布尔属性指示密码字段不可用于交互。此外，禁用的字段值不会与表单一起提交。
-
 ## 示例
 
 ### 申请社会安全号码
 
-此示例仅接受与有效的美国社会安全号码格式相匹配的输入。这些用于美国税务和识别目的的数字的格式为“123-45-6789”。还存在针对每个组中允许的值的各种规则。
+此示例仅接受与[有效的美国社会安全号码](https://en.wikipedia.org/wiki/Social_Security_number#Structure)格式相匹配的输入。这些用于美国税务和识别目的的数字的格式为“123-45-6789”。还存在针对每个组中允许的值的各种规则。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:195]:
```suggestion
此示例仅接受与[有效的美国社会安全号码](https://zh.wikipedia.org/zh-cn/社會安全號碼)格式相匹配的输入。这些用于美国税务和识别目的的数字的格式为“123-45-6789”。还存在针对每个组中允许的值的各种规则。
```

File: files/zh-cn/web/html/element/input/password/index.md [hunk 6]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
+  autocomplete="off" />
+<br />
+<label for="ssn">值：</label>
 <span id="current"></span>
 ```
 
-This uses a {{htmlattrxref("pattern", "input")}} which limits the entered value to strings representing legal Socal Security numbers. 很显然，这个正则表达式并不能保证有效的 SSN（因为我们没有进入社会保障局的数据库），但它确实保证数量可能是一个号; 它通常会避免无效的值。此外，它允许三组数字由空格，短划线（“ - ”）分隔，或者什么也不分。
-
-> **备注：** Isn't that regular expression just _beautiful_?
+此示例使用了 {{htmlattrxref("pattern", "input")}} 属性，限制了输入为一个合法的社会安全号码。很显然，这个正则表达式并不能保证有效的 SSN（因为我们没有访问到社会保障局的数据库），但它确实保证输入的值可能是合法值之一。它通常会避免无效的值。此外，它允许三组数字由空格，短划线（“-”）分隔，或者没有分隔符。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:215]:
```suggestion
此示例使用了 {{htmlattrxref("pattern", "input")}} 属性，限制了输入为一个合法的社会安全号码。很显然，这个正则表达式并不能保证有效的 SSN（因为我们没有访问到社会保障局的数据库），但它确实保证输入的值可能是合法值之一。它通常会避免无效的值。此外，它允许三组数字由空格、短划线（“-”）分隔，或者没有分隔符。
```

File: files/zh-cn/web/html/element/input/password/index.md [hunk 7]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
+{{htmlattrxref("inputmode", "input")}} 被设置为 `numeric`，以鼓励有虚拟键盘的设备切换到数字键盘布局，以方便输入。{{htmlattrxref("minlength", "input")}} 和 {{htmlattrxref("maxlength", "input")}}属性分别设置为 9 和 12，以要求值的长度至少为 9 个，不超过 12 个字符（前者在数字组之间没有分隔符，后者则有）。{{htmlattrxref("required", "input")}} 属性被用来表示该控件必须有一个值。最后，{{htmlattrxref("autocomplete", "input")}} 被设置为 `off`，以避免密码管理器和会话恢复功能试图设置其值，因为这根本就不是一个密码。
 
-#### JavaScript 版本的实例
+#### JavaScript
 
-这只是一些简单的代码，用于在屏幕上显示输入的 SSN，以便您可以看到它。显然这会破坏密码字段的目的，但它有助于试验模式。
+这只是一些简单的代码，用于在屏幕上显示输入的 SSN，以便您可以看到它。显然这会破坏密码字段的目的，但它有助于试验 `pattern`。
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:221]:
```suggestion
这只是一些简单的代码，用于在屏幕上显示输入的 SSN，以便你可以看到它。显然这会破坏密码字段的目的，但它有助于试验 `pattern`。
```

File: files/zh-cn/web/html/element/input/password/index.md [hunk 8]
```diff
@@ -5,205 +5,292 @@ slug: Web/HTML/Element/Input/password
...
+      <td>
+        {{htmlattrxref("autocomplete", "input")}}、
+        {{htmlattrxref("inputmode", "input")}}、
+        {{htmlattrxref("maxlength", "input")}}、
+        {{htmlattrxref("minlength", "input")}}、
+        {{htmlattrxref("pattern", "input")}}、
+        {{htmlattrxref("placeholder", "input")}}、
+        {{htmlattrxref("readonly", "input")}}、
+        {{htmlattrxref("required", "input")}} 和
+        {{htmlattrxref("size", "input")}}
```
Comment (yin1999) [inline files/zh-cn/web/html/element/input/password/index.md:264]: 不要换行

**Review by yin1999 — CHANGES_REQUESTED** [review]

File: files/zh-cn/web/html/element/input/password/index.md (hunk 8 above)
Comment (jasonren0403) [inline files/zh-cn/web/html/element/input/password/index.md:264]:
```suggestion
      <td>
        {{htmlattrxref("autocomplete", "input")}}、{{htmlattrxref("inputmode", "input")}}、{{htmlattrxref("maxlength", "input")}}、{{htmlattrxref("minlength", "input")}}、{{htmlattrxref("pattern", "input")}}、{{htmlattrxref("placeholder", "input")}}、{{htmlattrxref("readonly", "input")}}、{{htmlattrxref("required", "input")}} 和 {{htmlattrxref("size", "input")}}
```

**Review by yin1999 — APPROVED** [review]

**Pull request merged**